src/items/powerup_manager.hpp

```diff
@@ -126,30 +129,35 @@ class PowerupManager : public NoCopy
 
     /** For each powerup the weight (probability) used depending on the
      *  number of players. */
-    std::vector<int> m_weights[POSITION_COUNT];
+    std::vector<unsigned int> m_weights[POSITION_COUNT];
 
     /** A list of all powerups for a specific class. If a powerup
      *  has weight 5, it will be listed 5 times in this list, so
      *  randomly picking an entry from this for a position class will
      *  result in the right distribution of items. */
-    std::vector<PowerupType> m_powerups_for_position[POSITION_COUNT];
+    std::vector<PowerupType> m_powerups_for_reference_pos[POSITION_COUNT];
+
+    /** The mapping of each position to the corresponding position class. */
 
-    /** The mapping of each position to the corresponding position class.
-     *  There is one map for each different number of players, so it is
-     *  used like  m_position_to_class[number_players][position] */
-    std::vector<PositionClass> m_position_to_class;
+    std::vector<PositionClass> m_position_to_class_inf;
+    std::vector<PositionClass> m_position_to_class_sup;
+    std::vector<int> m_position_to_class_cutoff;
 
     PowerupType   getPowerupType(const std::string &name) const;
 
     void          loadWeights(const XMLNode &root,
+                              unsigned int num_karts,
                               const std::string &class_name,
                               PositionClass position_class);
+    void          updatePowerupClass(PowerupManager::PositionClass pos_class);
     PositionClass convertPositionToClass(unsigned int num_karts,
+                                         unsigned int position, bool class_sup = false);
+    unsigned int           convertPositionToClassWeight(unsigned int num_karts,
                                          unsigned int position);
 public:
                   PowerupManager  ();
                  ~PowerupManager  ();
-    void          loadAllPowerups ();
+    void          loadAllPowerups (unsigned int num_karts=10);
```

Comment:
why 10?

Reply:
I originally had this as a default fallback value, though the call of the function in powerup_manager.cpp should always specify the number of karts, so this may be not needed.

Reply:
If the arbitrary default value is never need, then indeed it's preferable to remove it